### DISCUSSION TITLE
No mapping properties for type when trying to write a file and using attributes.

### DISCUSSION THREAD

**Discussion by Temeos** [JoshClose/CsvHelper#1553] [Q&A]
I am trying to write into a csv file by using Attributes. 
I maybe should mension, that i am working in Unity. 
This is my code: 
`public class Log`
`{`
        `[Index(0)]`
        `[Name("Average Time")]`
   `    public float timeAverage;`
        `[Index(2)]`
        `[Name("Tries")]`
`       public float numberOfTries; `
        `[Index(1)]`
        `[Name("TPT")]`
`       public flaot averageTimePerTry; `
`}`

`public void MakeFile(Log log)`
`{`
   `using (var writer = new StreamWriter(Application.dataPath + "/Logs/logFile.csv"))`
`        using (var csv = new CsvWriter(writer, CultureInfo.InvariantCulture))`
 `       {        `
  `          csv.WriteRecord<Log>(log);`
   `     }`
`}`
So this gives me the error "No properties are mapped for type..." 

I also tried to use the ClassMap method and it worked, but i didn't have any headers. The code looked like this: 
` public class LogFileMap : ClassMap<Log>`
`    {`
 `       public LogFileMap()`
  `      {`
   `         Map(m => m.timeAverage).Index(0).Name("Average Time");`
    `        Map(m => m.numberOfTries).Index(1).Name("Number of tries");`
     `       Map(m => m.averageTimePerTry).Index(2).Name("Time on one Try");`
      `  }`
    `}`
with the addidtion of `csv.Configuration.RegisterClassMap<ClassMap>();` in the `MakeLog()` function.

If possible, i would like to know how to do this with the Attributes. But i will also be thankful for a fix in the ClassMap method. 

Best regards.

**Comment by remymantei** [comment]
Hi, did you find an answer to your issue ? Else, I recommand you to add getters and setters to your properties:

```
public class Log
{
    [Index(0)]
    [Name("Average Time")]
    public float TimeAverage { get; set; }

    [Index(2)]
    [Name("Tries")]
    public float NumberOfTries { get; set ; }

    [Index(1)]
    [Name("TPT")]
    public float AverageTimePerTry { get; set; }
}
```

Or use fluent ClassMap and remove Log class annotations but keep the getters and setters.

**Comment by yzorg** [comment]
This is configurable for CsvHelper, so if you **need** to keep them fields configure the writer to map fields, see [these comments on an issue from 2018](https://github.com/JoshClose/CsvHelper/issues/945#issuecomment-374091068).

But that said, I agree with @remymantei.  The convention for most .NET/C# code is: properties are serialized, while fields are not, even public fields. If possible it would be better to switch your code to follow the platform conventions.
